core/file_api.php

```diff
@@ -98,45 +98,83 @@ function file_get_display_name( $p_filename ) {
+ * @param array $p_bug_ids Array of bug ids
+ * @return void
  */
-function file_bug_attachment_count( $p_bug_id ) {
+function file_bug_attachment_count_cache( array $p_bug_ids ) {
 	global $g_cache_file_count;
 
-	# First check if we have a cache hit
-	if( isset( $g_cache_file_count[$p_bug_id] ) ) {
-		return $g_cache_file_count[$p_bug_id];
+	if( empty( $p_bug_ids ) ) {
+		return;
+	}
+
+	$t_ids_to_search = array();
+	foreach( $p_bug_ids as $t_id ) {
+		$c_id = (int)$t_id;
+		$t_ids_to_search[$c_id] = $c_id;
 	}
 
-	# If there is no cache hit, check if there is anything in
-	#   the cache. If the cache isn't empty and we didn't have
-	#   a hit, then there are not attachments for this bug.
-	if( count( $g_cache_file_count ) > 0 ) {
-		return 0;
+	db_param_push();
+	$t_params = array();
+	$t_in_values = array();
+	foreach( $t_ids_to_search as $t_id ) {
+		$t_params[] = (int)$t_id;
+		$t_in_values[] = db_param();
 	}
 
-	# Otherwise build the cache and return the attachment count
-	#   for the given bug (if any).
-	$t_query = 'SELECT bug_id, COUNT(bug_id) AS attachments FROM {bug_file} GROUP BY bug_id';
-	$t_result = db_query( $t_query );
+	$t_query = 'SELECT B.id AS bug_id, COUNT(F.bug_id) AS attachments'
+			. ' FROM {bug} B JOIN {bug_file} F ON ( B.id = F.bug_id )'
+			. ' WHERE B.id IN (' . implode( ',', $t_in_values ) . ')'
+			. ' GROUP BY B.id';
 
-	$t_file_count = 0;
+	$t_result = db_query( $t_query, $t_params );
 	while( $t_row = db_fetch_array( $t_result ) ) {
-		$g_cache_file_count[$t_row['bug_id']] = $t_row['attachments'];
-		if( $p_bug_id == $t_row['bug_id'] ) {
-			$t_file_count = $t_row['attachments'];
-		}
+		$c_bug_id = (int)$t_row['bug_id'];
+		$g_cache_file_count[$c_bug_id] = (int)$t_row['attachments'];
+		unset( $t_ids_to_search[$c_bug_id] );
 	}
 
-	# If no attachments are present, mark the cache to avoid
-	#   repeated queries for this.
-	if( count( $g_cache_file_count ) == 0 ) {
-		$g_cache_file_count['_no_files_'] = -1;
+	# set bugs without result to 0
+	foreach( $t_ids_to_search as $t_id ) {
+		$g_cache_file_count[$t_id] = 0;
 	}
+}
 
-	return $t_file_count;
+/**
+ * Check the number of attachments a bug has (if any)
+ * @param integer $p_bug_id A bug identifier.
+ * @return integer
+ */
+function file_bug_attachment_count( $p_bug_id ) {
+	global $g_cache_file_count;
+
+	# If it's not in cache, load the value
+	if( !isset( $g_cache_file_count[$p_bug_id] ) ) {
+		file_bug_attachment_count_cache( array( (int)$p_bug_id ) );
+	}
+
+	return $g_cache_file_count[$p_bug_id];
+}
+
+/**
+ * Clear a bug from the cache or all bugs if no bug id specified.
+ * @param integer $p_bug_id A bug identifier to clear (optional).
+ * @return boolean
+ * @access public
+ */
+function file_bug_attachment_count_clear_cache( $p_bug_id = null ) {
+	global $g_cache_file_count;
+
+	if( null === $p_bug_id ) {
+		$g_cache_file_count = array();
+	} else {
+		unset( $g_cache_file_count[(int)$p_bug_id] );
+	}
+
+	return true;
```

Comment:
The other clear cache methods used this scheme. 
No strong reason to return true, but a more detailed analysis on current usage, for the existing ones, should be done